zero/src/rpc/mod.rs

```diff
@@ -106,7 +105,8 @@ where
                     let block = cached_provider
                         .get_block((block_num as u64).into(), BlockTransactionsKind::Hashes)
                         .await
-                        .context("couldn't get block")?;
+                        .expect("could not retrieve block from provider")
+                        .ok_or(anyhow!("block not found {block_num}"))?;
```

Comment:
Why this change? We don't want to panic on `RpcError`?

Reply:
Good catch. Did not mean to keep the expect line there. Fixing